packages/hagrid/hagrid/cli.py

```diff
@@ -903,6 +903,22 @@ def check_gcloud_cli_installed() -> bool:
     return True
 
 
+def check_aws_cli_installed() -> bool:
+    try:
+        result = subprocess.run(  # nosec
+            ["aws", "--version"], stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT
+        )
+        if result.returncode != 0:
+            raise FileNotFoundError("AWS CLI not installed")
+    except Exception:  # nosec
+        msg = "\nYou don't appear to have the AWS CLI installed! \n\n\
+Please install it and then retry your command.\
+\n\nInstallation Instructions: https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html\n"
+        raise FileNotFoundError(msg)
+
+    return True
```

Comment:
You can use [`shutil.which`](https://docs.python.org/3/library/shutil.html#shutil.which) to check if `aws` is installed. Even better, implement the check by subclassing the `Dependency` class and use `BinaryInfo` which uses `shutil.which` underneath, similar to the check for `docker` https://github.com/OpenMined/PySyft/blob/d32b21a32e5e2e24d3b946d7fdec1a3f3bb7fef5/packages/hagrid/hagrid/deps.py#L157-L170.

Reply:
Thanks. I'm just following what was done for Azure to ensure consistency among the different cloud providers.
For instance, for Azure, we're doing:

https://github.com/OpenMined/PySyft/blob/c1bf1f64bae4c210f9646e560ef72bed17ed6786/packages/hagrid/hagrid/cli.py#L877-L890

Reply:
I see. Would be good to convert the azure check to the new method too for consistency, but that would be out of scope for this PR.

Reply:
Yup, that makes sense.

Reply:
Yeah I think we can handle this when we reboot hagrid which we really need to do.